crates/engine/tree/src/tree/mod.rs

```diff
@@ -565,16 +575,19 @@ where
     engine_kind: EngineApiKind,
     /// state root task thread pool
     state_root_task_pool: Arc<rayon::ThreadPool>,
+    /// Parallel prewarming thread pool
+    prewarming_thread_pool: rayon::ThreadPool,
```

Comment:
by the time we start SR, we should be (almost) entirely done with prewarming, right?

so I think we can simply reuse that pool?
I assume the workload distribution IO/CPU is similar to SR so even if we go with a different pool we can reuse that for both as well

Reply:
sure, can reuse the pool